lib/cdo/video/youtube.rb

```diff
@@ -0,0 +1,66 @@
+# This class downloads and processes a YouTube video by ID,
+# making it available for viewing through Code.org's fallback video player.
+
+# Dependency requirements:
+# viddl-rb (`gem install viddl-rb`) - for downloading the video
+# ffmpeg (`brew install ffmpeg` / `apt-get install ffmpeg`) - for transcoding local videos
+# s3_access_key_id and s3_secret_access_key defined (e.g. in `locals.yml`)
+
+require 'cdo/aws/s3'
+require 'tmpdir'
+require 'open-uri'
+
+class Youtube
+
+  VIDEO_BUCKET = 'videos.code.org'
+
+  # Process a video with a youtube identifier.
+  # If filename is provided, transcode the local file with ffmpeg.
+  # If no filename is provided, download the transcoded video from YouTube.
+  #
+  # When downloading from YouTube, an HTTP head request will first check the absence of the file.
+  # If `force`==true, the head request will be skipped.
+  def self.process(id, filename=nil, force=false)
+    if filename.nil? && !force
+      require 'httparty'
+      thumbnail_url = "https://#{VIDEO_BUCKET}/youtube/#{id}.jpg"
+      response = HTTParty.head(thumbnail_url).response
+      if response.is_a? ::Net::HTTPSuccess
+        puts "Video id: #{id} already processed"
```

Comment:
We have a mix of `$log`, `Pegasus.logger`, `Rails.logger`, `slog`, and `puts` in use throughout the codebase, with `puts` often used in small scripts and basic standard output, but no guidelines on which logger to use where and why. Last month, I added a [simple logger implementation](https://github.com/code-dot-org/code-dot-org/commit/058868917f844e7e0ec0331a624c064e20fe0963#diff-63ce6b57dd8496d4e8739cf50dd128e7R179) to `CDO.log`, since we had no other standard-out logger available across our sub-environments for use in non-Pegasus/Rails scripts. This logger's functionality is currently no different from `puts`, but for now I've changed these new lines to use `CDO.log`- I'd love to see someone eventually have time to work on cleaning up/standardizing our logging practices throughout our codebase.